core/src/main/java/org/wso2/msf4j/ServiceMethodInfo.java

```diff
@@ -44,6 +47,10 @@ public Method getMethod() {
         return method;
     }
 
+    public Object[] getArgs() {
+        return Arrays.copyOf(args, args.length);
```

Comment:
Is it absolutely necessary to create a copy of the array here? If not better not to do it as it may affect memory consumption.